src/main/java/com/walabot/home/ble/sdk/BleApiHandler.kt

```diff
@@ -89,8 +87,9 @@ private fun EspBleApi.pair(host: String) {
 private fun EspBleApi.performParingWithCloud(host: String, code: String?) {
     callback.onResult(Result(EspPairingEvent.StagePairWithCloud), this)
     code?.let { it1 ->
-        Connection().pairing(code, cloudCredentials.idToken!!) {
+        Connection().pairing(code, config.token!!) {
             if (it.isSuccess) {
+                deviceId = it.getOrNull()?.get("deviceId") as? String
```

Comment:
I replaced the devInfo to hashmap for be able to create custom made devInfo from the bridge point of view, you should get a map (which will be converted to string) 
I can't create a custom devInfo instance because it's a protobuf class, so if the end result is map I send a map from the sdk